src/handler.rs

```diff
@@ -289,6 +334,11 @@ impl ObsJobHandler {
 
     #[instrument(skip(self))]
     async fn run_dput(&mut self, args: DputAction) -> Result<()> {
+        let mut artifacts = GitLabArtifacts {
+            job: &self.job,
+            artifacts: &mut self.artifacts,
+        };
```

Comment:
As this seems to be done a few time would it make sense to implement Into? That or have a inner structure encapsulating it to avoid the conversion